src/error.rs

```diff
@@ -46,6 +47,10 @@ pub enum Error {
     /// An error from the `tungstenite` crate.
     #[cfg(feature = "gateway")]
     Tungstenite(Box<TungsteniteError>),
+    /// An error from the [`secrets`] module.
+    ///
+    /// [`secrets`]: crate::secrets
+    Token(TokenError),
```

Comment:
Does this have to be a part of this enum? I can't see a spot that returns this instead of just `Result<T, TokenError>`.

Reply:
I figured this would be useful when using `?` so that `TokenError` could coerce to `Error`.